### PR TITLE
Add executed messages

### PR DESCRIPTION
**Pull request by AStox**
Adds the following messages: ExecutedDecreaseInvestOrder, ExecutedDecreaseRedeemOrder, ExecutedCollectInvest, and ExecutedCollectRedeem.

### CODE REVIEW

**Review by offerijns — COMMENTED** [review]
Build needs to be fixed, but message params look good to me

**Comment by AStox** [comment]
Yah not sure why the build is failing, the draft PR was building, and it all builds/tests locally. I'm gonna come back to this in an hour or so in case its just a github CI issue

**Comment by AStox** [comment]
The issue ended up being that `Connectors.sol` was over the 25kb size limit, but fixing that actually led me to some issues with our github workflows/profiles in general, so all thats fixed now.

File: src/routers/Gateway.sol [hunk 1]
```diff
@@ -22,6 +22,40 @@ interface ConnectorLike {
...
+        uint128 currency,
+        uint128 currencyPayout,
+        uint128 remainingInvestOrder
+    ) external;
+    function handleExecutedDecreaseRedeemOrder(
+        uint64 poolId,
+        bytes16 trancheId,
+        bytes32 investor,
+        uint128 currency,
+        uint128 trancheTokenPayouts,
```
Comment (offerijns) [inline src/routers/Gateway.sol:38]: Super minor nit but to make it consistent with the param in `handleExecutedCollectInvest` and the messages
```suggestion
        uint128 trancheTokensPayouts,
```

File: src/Connector.sol [hunk 2]
```diff
@@ -334,4 +334,50 @@ contract CentrifugeConnector {
...
+    ) public onlyGateway {
+        // TODO: Implement
+    }
+
+    function handleExecutedDecreaseRedeemOrder(
+        uint64 poolId,
+        bytes16 trancheId,
+        bytes32 investor,
+        uint128 currency,
+        uint128 trancheTokenPayouts,
```
Comment (offerijns) [inline src/Connector.sol:354]:
```suggestion
        uint128 trancheTokensPayout,
```

File: src/Messages.sol [hunk 3]
```diff
@@ -558,6 +566,186 @@ library ConnectorMessages {
...
+
+    function parseExecutedDecreaseRedeemOrder(bytes29 _msg)
+        internal
+        pure
+        returns (
+            uint64 poolId,
+            bytes16 trancheId,
+            bytes32 investor,
+            uint128 currency,
+            uint128 trancheTokenPayouts,
```
Comment (offerijns) [inline src/Messages.sol:643]:
```suggestion
            uint128 trancheTokensPayout,
```

File: src/Messages.sol [hunk 4]
```diff
@@ -558,6 +566,186 @@ library ConnectorMessages {
...
+            uint128 remainingRedeemOrder
+        )
+    {
+        poolId = uint64(_msg.indexUint(1, 8));
+        trancheId = bytes16(_msg.index(9, 16));
+        investor = bytes32(_msg.index(25, 32));
+        currency = uint128(_msg.indexUint(57, 16));
+        trancheTokenPayouts = uint128(_msg.indexUint(73, 16));
```
Comment (offerijns) [inline src/Messages.sol:651]:
```suggestion
        trancheTokensPayout = uint128(_msg.indexUint(73, 16));
```

File: src/routers/Gateway.sol [hunk 5]
```diff
@@ -275,6 +309,56 @@ contract ConnectorGateway {
...
+            connector.handleExecutedDecreaseInvestOrder(
+                poolId, trancheId, investor, currency, currencyPayout, remainingInvestOrder
+            );
+        } else if (ConnectorMessages.isExecutedDecreaseRedeemOrder(_msg)) {
+            (
+                uint64 poolId,
+                bytes16 trancheId,
+                bytes32 investor,
+                uint128 currency,
+                uint128 trancheTokenPayouts,
```
Comment (offerijns) [inline src/routers/Gateway.sol:330]:
```suggestion
                uint128 trancheTokensPayout,
```

File: src/routers/Gateway.sol [hunk 6]
```diff
@@ -275,6 +309,56 @@ contract ConnectorGateway {
...
+                uint128 remainingRedeemOrder
+            ) = ConnectorMessages.parseExecutedDecreaseRedeemOrder(_msg);
+            connector.handleExecutedDecreaseRedeemOrder(
+                poolId, trancheId, investor, currency, trancheTokenPayouts, remainingRedeemOrder
```
Comment (offerijns) [inline src/routers/Gateway.sol:334]:
```suggestion
                poolId, trancheId, investor, currency, trancheTokensPayout, remainingRedeemOrder
```

**Review by offerijns — COMMENTED** [review]
Nice to see the foundry profile cleanups!

**Comment by AStox** [comment]
Nice catch! I didn't even notice I had named them differently tbh

File: src/Messages.sol [hunk 7]
```diff
@@ -39,7 +39,15 @@ library ConnectorMessages {
...
         /// 14 - Collect Redeem
         CollectRedeem,
-        /// 15 - Add a new admin
+        /// 15 - Executed Decrease Invest Order
+        ExecutedDecreaseInvestOrder,
+        /// 16 - Executed Decrease Redeem Order
+        ExecutedDecreaseRedeemOrder,
+        /// 17 - Executed Collect Invest
+        ExecutedCollectInvest,
+        /// 18 - Executed Collect Redeem
```
Comment (NunoAlexandre) [inline src/Messages.sol:48]: I wonder if we would be better off removing the text in these comments

File: src/Connector.sol [hunk 8]
```diff
@@ -334,4 +334,50 @@ contract CentrifugeConnector {
...
+
+    function handleExecutedDecreaseInvestOrder(
+        uint64 poolId,
+        bytes16 trancheId,
+        bytes32 investor,
+        uint128 currency,
+        uint128 currencyPayout,
+        uint128 remainingInvestOrder
+    ) public onlyGateway {
+        // TODO: Implement
```
Comment (NunoAlexandre) [inline src/Connector.sol:346]: Do we still want to implement this here or leave it for a follow up PR when Alina's Pr is merged? Or base this one off Alina's and do it right away?

Comment (ilinzweilin) [reply src/Connector.sol:346]: yeah, I agree. Can be removed :) I have added the implementation in my pr

File: src/routers/Gateway.sol [hunk 9]
```diff
@@ -275,6 +309,56 @@ contract ConnectorGateway {
...
+            connector.handleExecutedDecreaseRedeemOrder(
+                poolId, trancheId, investor, currency, trancheTokensPayout, remainingRedeemOrder
+            );
+        } else if (ConnectorMessages.isExecutedCollectInvest(_msg)) {
+            (
+                uint64 poolId,
+                bytes16 trancheId,
+                bytes32 investor,
+                uint128 currency,
+                uint128 currencyPayout,
```
Comment (ilinzweilin) [inline src/routers/Gateway.sol:342]: currencyInvested instead of currencyPayout

**Review by ilinzweilin — APPROVED** [review]
LGTM, just left some minor comments.
Can be merged after fixing build

**Comment by NunoAlexandre** [comment]
Let's not merge this before testing the encoding/decoding of these messages as advised [here](https://github.com/centrifuge/connectors/pull/70#pullrequestreview-1500151392) please 🙏

File: src/routers/Gateway.sol (hunk 9 above)
Comment (offerijns) [reply src/routers/Gateway.sol]: I think we wanted the message parameters to be consistent between cent chain and solidity, so would not rename this here

Comment (offerijns) [reply src/routers/Gateway.sol]: There were long discussions about these naming decisions in https://github.com/centrifuge/centrifuge-chain/pull/1363

**Review by NunoAlexandre — APPROVED** [review]
LGTM :)

**Pull request merged**